### ISSUE TITLE
LargeBinary has a required field with a default value?

### ISSUE THREAD

**Issue by johnthagen** [collerek/ormar#186]
**Describe the bug**

`LargeBinary` has a required field, `max_length` but it has a default value. This means that IDE's can't help a user know that they forget a parameter.

```py
class LargeBinary(ModelFieldFactory, bytes):
    """
    LargeBinary field factory that construct Field classes and populated their values.
    """

    _type = bytes
    _sample = "bytes"

    def __new__(  # type: ignore # noqa CFQ002
        cls, *, max_length: int = None, **kwargs: Any
    ) -> BaseField:  # type: ignore
```

**To Reproduce**

```py
class MyModel(Model):
    class Meta:
        abstract = True
        metadata = metadata
        database = database

    data: bytes = LargeBinary()  # IDE cannot give warning here.
```

**Expected behavior**

I believe the class should be defined as:

```py
class LargeBinary(ModelFieldFactory, bytes):
    """
    LargeBinary field factory that construct Field classes and populated their values.
    """

    _type = bytes
    _sample = "bytes"

    def __new__(  # type: ignore # noqa CFQ002
        cls, *, max_length: int, **kwargs: Any
    ) -> BaseField:  # type: ignore
```

**An aside**

It seems like not all databases require a `max_length`. Could another option be that the parameter is optional, and only used if the underlying database needs it (such as MySQL)?

**Comment by collerek** [comment]
Yes, the default value should be removed.

The problem is that at the point of validation the field is not yet assigned to the model hence it does not "know" what dialect is used in the model. And it would only fail during creation of db/table that can be harder to debug/find cause, and I want to fail early (zen of python :) ) The same applies to String, not all backends require length but it's mandatory in `ormar`.

Will fix that in next release.

**Comment by johnthagen** [comment]
Sounds good. This also led me to do some research and learn that SQLite does have a global max of 1GB by default for any BLOB field: https://www.sqlite.org/limits.html

**Comment by collerek** [comment]
closed in #195
